File: andkit/andkit-rpc/src/main/java/com/anymore/andkit/rpc/converter/HuskWith.java
package com.anymore.andkit.rpc.converter;

import com.anymore.andkit.rpc.ResponseWrapper;

/**
 * 这个类用kotlin写就编译失败
 * Created by anymore on 2022/4/2.
 */
public @interface HuskWith {

    @SuppressWarnings("rawtypes")
    Class<? extends ResponseWrapper> value();

    long[] codes() default {0};
}

File: andkit/andkit-rpc/src/main/java/com/anymore/andkit/rpc/converter/HuskWith.kt
package com.anymore.andkit.rpc.converter

import com.anymore.andkit.rpc.ResponseWrapper
import kotlin.reflect.KClass

/**
 * 注意：HuskWith只接受Class<out ResponseWrapper>
 * Created by anymore on 2022/3/30.
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class HuskWith(val value: KClass<out ResponseWrapper<*>>, val codes: LongArray = [0L])
